### PR TITLE
Update react-server to write server side timings/logs to the response document.

### PR DESCRIPTION
**Pull request by vinsewah**
See: https://github.com/redfin/react-server/issues/272

### CODE REVIEW

**Comment by doug-wade** [comment]
@vinsewah your build failed with some linting errors

```
   1:5   error  'common' is defined but never used   no-unused-vars
   2:19  error  Unexpected use of process.env        no-process-env
   9:2   error  Expected space(s) after "if"         keyword-spacing
  40:3   error  Expected space(s) after "if"         keyword-spacing
  68:2   error  Expected space(s) after "if"         keyword-spacing
  68:11  error  Expected '===' and instead saw '=='  eqeqeq
  71:7   error  Expected space(s) after "if"         keyword-spacing
  71:16  error  Expected '===' and instead saw '=='  eqeqeq
  79:61  error  Duplicate key 'TimeResponseLogger'   no-dupe-keys
✖ 9 problems (9 errors, 0 warnings)
```

You can verify that they've been resolved locally with an `npm test` in the monorepo root, or a `gulp eslint` in packages/react-server

**Comment by gigabo** [comment]
@vinsewah This is cool!

Collecting all of the log output in the queue could get a little heavy in terms of GC churn.

Maybe better to skip putting anything _into_ the queue `if !req.query._debug_output_logs`?

File: packages/react-server/core/logging/response.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+var	flushLogsToResponse = function(res) {
+	if (queue().length > 0) {
+		res.write("<script>");
+		res.write(`window.reactServerLogs = ${JSON.stringify(queue())};\n`);
+		res.write("</script>");
+	}
+}
+
+var setResponseLoggerPage = function(page) {
+	if (RLS.isActive() || !!page) {
```
Comment (gigabo) [inline packages/react-server/core/logging/response.js]: Should this be an `&&`?

File: packages/react-server/core/logging/response.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	else if (group === "gauge") {
+		return new GaugeResponseLogger(opts);
+	}
+	else {
+		return new ResponseLogger(opts);
+	}
+}
+
+var	flushLogsToResponse = function(res) {
```
Comment (gigabo) [inline packages/react-server/core/logging/response.js]: Looks like some funny whitespace snuck in here after the `var`.

File: packages/react-server/core/logging/response.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	log(level, msg, meta, callback) {
+		var tuple = [
+			this.module,
+			msg,
+			meta[this.key],
+			this.lastModuleToken,
+		];
+		pushToQueue(tuple);
```
Comment (gigabo) [inline packages/react-server/core/logging/response.js]: Let's make this even _lower_ overhead by skipping tuple creation and `pushToQueue` call `if !RLS().doLog`.

Really want to be doing as little as possible when not enabled.

File: packages/react-server/package.json [hunk 4]
```diff
@@ -34,7 +34,7 @@
     "mobile-detect": "1.3.0",
     "q": "1.4.1",
     "querystring": "0.2.0",
-    "request-local-storage": "1.0.0",
+    "request-local-storage": "1.1.0",
```
Comment (gigabo) [inline packages/react-server/package.json]: I think this part already landed in #617.  A rebase should get it out of the diff here.

**Pull request merged**